packages/expo-doctor/src/utils/explainDependencies.ts

```diff
@@ -0,0 +1,132 @@
+import spawnAsync, { SpawnResult } from '@expo/spawn-async';
+import chalk from 'chalk';
+import semver from 'semver';
+
+import { RootNodePackage, VersionSpec } from './explainDependencies.types';
+
+type TargetPackage = { name: string; version?: VersionSpec };
+
+function isSpawnResult(result: any): result is SpawnResult {
+  return 'stderr' in result && 'stdout' in result && 'status' in result;
+}
+
+/** Spawn `npm explain [name] --json` and return the parsed JSON. Returns `null` if the requested package is not installed. */
+async function explainAsync(
```

Comment:
Might be worth adding some tests for this which mock spawn and return some static results.